src/ansi-c/ansi_c_entry_point.cpp

```diff
@@ -42,7 +42,8 @@ exprt::operandst build_function_environment(
       base_name,
       p.type(),
       p.source_location(),
-      object_factory_parameters);
+      object_factory_parameters,
+      lifetimet::AUTOMATIC_LOCAL);
```

Comment:
Random place for comment: it would be nice if the commit message could explain _why_ the existing symbol factory wasn't usable as-is.